com.unity.netcode.gameobjects/Runtime/Core/NetworkObject.cs

```diff
@@ -1051,36 +1144,48 @@ public unsafe void Deserialize(FastBufferReader reader)
                     throw new OverflowException("Could not deserialize SceneObject: Out of buffer space.");
                 }
                 reader.ReadValue(out Header);
-                var readSize = Header.HasParent ? FastBufferWriter.GetWriteSize(ParentObjectId) : 0;
-                readSize += Header.HasTransform ? FastBufferWriter.GetWriteSize(Transform) : 0;
-                readSize += Header.IsReparented ? FastBufferWriter.GetWriteSize(IsLatestParentSet) + (IsLatestParentSet ? FastBufferWriter.GetWriteSize<ulong>() : 0) : 0;
+                var readSize = 0;
+                if (Header.HasParent)
+                {
+                    readSize += FastBufferWriter.GetWriteSize(ParentObjectId);
+                    readSize += FastBufferWriter.GetWriteSize(WorldPositionStays);
+                    readSize += FastBufferWriter.GetWriteSize(IsLatestParentSet);
+                    // We need to read at this point in order to get the IsLatestParentSet value
+                    if (!reader.TryBeginRead(readSize))
+                    {
+                        throw new OverflowException("Could not deserialize SceneObject: Out of buffer space.");
```

Comment:
Nit: I'd make this error something like "Trying to read past the end of the buffer" rather than "out of buffer space". "Out of buffer space" could be interpreted as an Out Of Memory issue (i.e., couldn't allocate a big enough buffer to read it).

Reply:
Sure, I can adjust that language.
_(I actually just used the same message if trying to read the header failed.)_